### ISSUE TITLE
[rlgl] Geometry Shader support

### ISSUE THREAD

**Issue by Crydsch** [raysan5/raylib#1575]
Hey all,
i've been experimenting with geometry shaders and adding support for them to raylib.

## How
To do this i changed the following functions in rlgl.h and ralib.h
```c
RLAPI Shader LoadShader(const char *vsFileName, const char *fsFileName, const char *gsFileName);
RLAPI Shader LoadShaderCode(const char* vsCode, const char* fsCode, const char* gsCode);
static unsigned int LoadShaderProgram(unsigned int vShaderId, unsigned int fShaderId, unsigned int gShaderId);
```

and added `unsigned int defaultGShaderId;` to `RLGL.State`
Since a pass-through geometry shader still imposes a considerable overhead i opted to not implement one in the default shader.
Instead `defaultGShaderId ` is never set to anything else than 0 and is used to check wether a geometry shader is in use at all.
If `gsFileName `(or `gsCode`) is NULL, we only attach the vertex and fragment shader to the shader program (or just use the default one).

## Considerations
- Breaking Change: function calls
  It seems C does not really support function overloads, so i had to replace them.
  Maybe use something like `LoadShaderEx(vs,fs,gs) `and keep `LoadShader(vs,gs)`?
- OpenGl support:
  Geometry Shader are core since version 3.2
  Instancing and Output streams are core since 4.0
  This would make them available in raylib, but only with GRAPHICS_API_OPENGL_33
  Source: https://www.khronos.org/opengl/wiki/Geometry_Shader
- Performance:
  Turns out geometry shader have notorious bad performance and their usefullness is highly dependent on the situation (and hardware).
  Sources:
   Often referenced post, seems still uptodate: http://www.joshbarczak.com/blog/?p=667
   Factorio dev article, with some testing: https://factorio.com/blog/post/fff-251

## Bottom line
I had fun experimenting on this and it works, but i have chosen not to use them in my project.
If you are interested i can send a PR.
If for nothing else this issue serves as info for others.

**Comment by raysan5** [comment]
@Crydsch I'm afraid I'm not adding this feature to raylib, it seems to have very specific use cases...

In any case, thank you very much for the detailed documentation, I can be useful for other users looking to implement it.
